### ISSUE TITLE
Cannot open scenarioB from scenarioA

### ISSUE THREAD

**Issue by johnsickels** [flocasts/flagpole#20]
## Expected Behavior
In previous versions I was able to open a scenario from within another scenario. This allowed me to build something in the browser and delete it with an api scenario and assert it's success, or perform a `.html` smoke test, get some data from it and pass it to another scenario's `.open` route. 

## Actual Behavior
The finalUrl does not accurately reflect what was passed in .open()

## Steps to Reproduce the Problem

```
import { Flagpole } from "flagpole";

const suite = Flagpole.suite("Mixing Browser and API scenarios").base(
  "https://www.milesplit.com/"
);

suite
  .browser("Homepage Loads")
  .open("/")
  .next(async (context) => {
    await context.exists("#content .coverStory");
    apiCall.open("/api/v1/rankings");
    anotherPage.open("/results");
  });

const apiCall = suite.json("Simple API assertion").next(async (context) => {
  context.comment(context.response.finalUrl);
});

const anotherPage = suite
  .browser("Opening another page from scenario A")
  .next(async (context) => {
    context.comment(context.response.finalUrl);
  });
```

finalUrl in the `.json` scenario is *https://www.milesplit.com* as it _should_ be *https://www.milesplit.com/api/v1/rankings* or *https://www.milesplit.com/results*

## Specifications

  - Version: 2.2.9

**Comment by jasonbyrne** [comment]
Thanks for the details. Looking into it now.

**Comment by jasonbyrne** [comment]
See when I get a simple reproduction case, it's easy to track down! This is fixed in 2.2.10

The problem was with this line:
https://github.com/flocasts/flagpole/commit/83c2dbcd02a6e60bf61104c66af02c1ce5d3d7de#diff-5e71c7c45c9fd79b72c638687e0cfa26L218

I was setting a default input URI as "/" and so the other scenarios were executing right away because they had a URL set.
